File: server/src/main/scala/producer/message_generator/data_generators/bytes/BytesFromBase64Generator.scala
package producer.message_generator.data_generators.bytes

import com.tools.teal.pulsar.ui.producer.v1.producer as pb
import producer.message_generator.data_generators.string.StringGenerator
import java.util.Base64
import org.graalvm.polyglot.Context

case class BytesFromBase64Generator(
    generator: StringGenerator
):
    def generate(polyglotContext: Context): Array[Byte] =
        val base64String = generator.generate(polyglotContext)
        Base64.getDecoder.decode(base64String)

object BytesFromBase64Generator:
    def fromPb(v: pb.BytesFromBase64Generator): BytesFromBase64Generator =
        BytesFromBase64Generator(
            generator = StringGenerator.fromPb(v.generator.get)
        )

    def toPb(v: BytesFromBase64Generator): pb.BytesFromBase64Generator =
        pb.BytesFromBase64Generator(
            generator = Some(StringGenerator.toPb(v.generator))
        )

File: ui/components/ui/ProducerSession/producer-task/message-generator/data-generators/bytes/bytes-from-base64-generator.ts
import * as pb from '../../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';
import { StringGenerator, stringGeneratorFromPb, stringGeneratorToPb } from '../string-generator/string-generator';

export type BytesFromBase64Generator = {
  type: 'bytes-from-base64-generator',
  generator: StringGenerator
};

export function bytesFromBase64GeneratorFromPb(v: pb.BytesFromBase64Generator): BytesFromBase64Generator {
  return {
    type: 'bytes-from-base64-generator',
    generator: stringGeneratorFromPb(v.getGenerator()!)
  };
}

export function bytesFromBase64GeneratorToPb(v: BytesFromBase64Generator): pb.BytesFromBase64Generator {
  const message = new pb.BytesFromBase64Generator();
  message.setGenerator(stringGeneratorToPb(v.generator));

  return message;
}
